framework/tlvf/src/src/CmduMessageRx.cpp

```diff
@@ -31,12 +71,145 @@ CmduMessageRx::~CmduMessageRx()
     }
 }
 
-/*
- * TODO
- * change all pointer arguments to const type where applicable
- */
+std::shared_ptr<BaseClass> CmduMessageRx::parseWscTlv()
+{
+    Tlv *tlv = (Tlv *)getClassVector().back()->getBuffPtr();
+
+    if (eTlvType(tlv->type()) != eTlvType::TLV_WSC)
+        return nullptr;
+
+    Wsc *wsc = (Wsc *)(tlv->data());
+    uint16_t type = wsc->type();
+    while (type != WSC::ATTR_MSG_TYPE) {
+        wsc = wsc->next();
+        type = wsc->type();
+    }
+    if (*wsc->data() == WSC::WSC_MSG_TYPE_M2)
+        return addClass<tlvWscM2>();
+    else
+        return addClass<tlvWscM1>();
+
+    return nullptr;
+}
 
-std::shared_ptr<cCmduHeader> CmduMessageRx::parse(uint8_t *buff, size_t buff_len, bool swap_needed)
+std::shared_ptr<BaseClass> CmduMessageRx::parseNextTlv()
```

Comment:
Did you manually write this method.........?

Reply:
No, @Galvien updated tlvf in a different commit which generates the switch statement. I think however that since it is not bound to be changed too often, that we can have it here statically instead of generating it each time.

Reply:
I think that's a bad practice to insert auto generated code into non-auto generated files.
A better approach would be exporting this logic to an external class (which would be autogenerated), and have this class as a member of CmduMessageRx. That way, "parseNextTlv" would only call this class' method.

Reply:
Fixed in a big fixup splitting to new classes and files. We planned to do that anyway but since you asked so nicely.... :)